internal/controllers/rbac.go

```diff
@@ -0,0 +1,312 @@
+
+		// Update labels and annotations managed by the operator
+		for key, val := range labels {
+			metav1.SetMetaDataLabel(&sa.ObjectMeta, key, val)
+		}
+		for key, val := range annotations {
+			metav1.SetMetaDataAnnotation(&sa.ObjectMeta, key, val)
+		}
+
+		// Set the Cryostat CR as controller
+		if err := controllerutil.SetControllerReference(owner, sa, r.Scheme); err != nil {
+			return err
+		}
+		// AutomountServiceAccountToken specified in Pod, which takes precedence
+		// Secrets, ImagePullSecrets are modified by Kubernetes/OpenShift
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	r.Log.Info(fmt.Sprintf("Service Account %s", op), "name", sa.Name, "namespace", sa.Namespace)
+	return nil
+}
+
+func (r *CryostatReconciler) createOrUpdateRole(ctx context.Context, role *rbacv1.Role,
+	owner metav1.Object, rules []rbacv1.PolicyRule) error {
+	op, err := controllerutil.CreateOrUpdate(ctx, r.Client, role, func() error {
+		// Update the list of PolicyRules
+		role.Rules = rules
+
+		// Set the Cryostat CR as controller
+		if err := controllerutil.SetControllerReference(owner, role, r.Scheme); err != nil {
+			return err
+		}
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	r.Log.Info(fmt.Sprintf("Role %s", op), "name", role.Name, "namespace", role.Namespace)
+	return nil
+}
+
+func (r *CryostatReconciler) createOrUpdateRoleBinding(ctx context.Context, binding *rbacv1.RoleBinding,
+	owner metav1.Object, subjects []rbacv1.Subject, roleRef *rbacv1.RoleRef) error {
+	op, err := controllerutil.CreateOrUpdate(ctx, r.Client, binding, func() error {
+		// Update the list of Subjects
+		binding.Subjects = subjects
+		// Update the Role reference
+		binding.RoleRef = *roleRef
+
+		// Set the Cryostat CR as controller
+		if err := controllerutil.SetControllerReference(owner, binding, r.Scheme); err != nil {
+			return err
+		}
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	r.Log.Info(fmt.Sprintf("Role Binding %s", op), "name", binding.Name, "namespace", binding.Namespace)
+	return nil
+}
+
+func (r *CryostatReconciler) createOrUpdateClusterRoleBinding(ctx context.Context, binding *rbacv1.ClusterRoleBinding,
+	owner metav1.Object, subjects []rbacv1.Subject, roleRef *rbacv1.RoleRef) error {
+	op, err := controllerutil.CreateOrUpdate(ctx, r.Client, binding, func() error {
+		// Update the list of Subjects
+		binding.Subjects = subjects
+		// Update the Role reference
+		binding.RoleRef = *roleRef
+
+		// ClusterRoleBinding can't be owned by namespaced CR, clean up using finalizer
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	r.Log.Info(fmt.Sprintf("Cluster Role Binding %s", op), "name", binding.Name, "namespace", binding.Namespace)
+	return nil
+}
+
+func (r *CryostatReconciler) deleteClusterRoleBinding(ctx context.Context, cr *operatorv1beta1.Cryostat) error {
+	clusterBinding := newClusterRoleBinding(cr) // TODO move to finalizeRBAC, add tests
```

Comment:
Is this still a `TODO`?

Reply:
Nope, thanks for noticing that. I've removed it.